lib/Stats.tsx

```diff
@@ -1,101 +1,236 @@
+        const tracks = pipeline.tracks?.map((track, index) =>
+          Object.assign({ index }, track),
+        )
+        const videoTrack = tracks?.find((track) => track.type === 'video')
+        if (videoTrack !== undefined) {
+          const { coding, profile, level } = videoTrack?.codec
+          const framerate = Number(
+            pipeline.framerate[videoTrack.index].toFixed(2),
+          )
+          const bitrate = Math.round(pipeline.bitrate[videoTrack.index] / 1000)
+
+          statsData = statsData.concat([
+            {
+              name: 'Encoding',
+              value: `${coding} ${profile} (${level})`,
+            },
+            {
+              name: 'Frame rate',
+              value: framerate,
+              unit: 'fps',
+            },
+            {
+              name: 'Bitrate',
+              value: bitrate,
+              unit: 'kbit/s',
+            },
+          ])
+        }
       }
-      return 'Motion JPEG'
+
+      setStats(statsData)
     }
 
-    return 'Unknown'
-  }, [api, parameters])
+    updateValues()
+
+    statsUpdateRef.current = window.setInterval(updateValues, 1000)
 
-  const resolution = useMemo(() => {
-    const { width, height } = videoProperties
-    return `${width}x${height}`
-  }, [videoProperties])
+    return () => window.clearInterval(statsUpdateRef.current)
+  }, [api, parameters, videoProperties, refresh, setStats])
 
   return (
-    <StatsWrapper>
-      <StatsHeader>Stream statistics</StatsHeader>
-      <StatContainer>
-        <Stat>
-          <Title>Host</Title>
-          <div> {host}</div>
-        </Stat>
-        <Stat>
-          <Title>Format</Title>
-          <div> {streamType}</div>
-        </Stat>
-        <Stat>
-          <Title>Resolution</Title>
-          <div> {resolution}</div>
-        </Stat>
-        <Stat>
-          <Title>Refreshes</Title>
-          <div> {refresh}</div>
-        </Stat>
-      </StatContainer>
-    </StatsWrapper>
+    <>
+      {showStats ? (
+        <StatsWrapper>
+          <StatsHeader>
+            <StatsIcon clickable={false}>
+              <StreamStats />
+            </StatsIcon>
+            <StatsTitle>Client stream data</StatsTitle>
+            <StatsHide>
+              <HideLink href="" onClick={onToggleStats}>
+                Hide
+              </HideLink>
+            </StatsHide>
+          </StatsHeader>
+          <StatsData>
+            {stats.length > 0
+              ? stats.map((stat) => {
+                  return (
+                    <StatItem key={stat.name}>
+                      <StatName>{stat.name}</StatName>
+                      <StatValue>{`${stat.value} ${
+                        stat.unit !== undefined ? stat.unit : ''
+                      }`}</StatValue>
+                    </StatItem>
+                  )
+                })
+              : null}
+          </StatsData>
+        </StatsWrapper>
+      ) : (
+        <StatsShow>
```

Comment:
It annoys me that this is 100% opacity, but it's 88% when open.

Reply:
It also annoys me that the icon shift position between open and closed.

Reply:
Fixed #1, #2 is by design again : )